webapp/components/sidebar.jsx

```diff
@@ -121,9 +121,9 @@ export default class Sidebar extends React.Component {
                     name: Utils.getDirectChannelName(currentUserId, teammateId),
                     last_post_at: 0,
                     total_msg_count: 0,
-                    type: Constants.DM_CHANNEL,
-                    fake: true
+                    type: Constants.DM_CHANNEL
                 };
+                Utils.openDirectChannelToUser(UserStore.getProfile(teammateId), null, null);
```

Comment:
If this is to fix PLT-3708, it will only automatically create DM channels for the users in your sidebar when you first sign up. You'll still get an error when you try to open a DM channel with other users not in that list already. Even if that weren't the case, I don't think it should be done here it's an unexpected side effect of a function that's supposed to get the state of an object.

Since PLT-3708 is specifically for the channel switcher, it would probably make more sense for `SwitchChannelModal.handleSubmit` to construct a channel object with `channel.fake = true` and send it to `ChannelActions.goToChannel` so that it's super clear when and why it's creating a new direct channel.